appinventor/components/src/com/google/appinventor/components/runtime/util/FroyoUtil.java

```diff
@@ -153,6 +153,11 @@ public void onPageStarted(WebView view, String url, Bitmap favicon) {
       public void onPageFinished(WebView view, String url) {
         EventDispatcher.dispatchEvent(component, "PageLoaded", url);
       }
+
+      @Override
+      public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
+        EventDispatcher.dispatchEvent(component, "ErrorOccurred", errorCode, description, failingUrl);
```

Comment:
The Android documentation doesn't say one way or another whether this will be called on the main thread for errors not occurring on the main thread, so it would be safer to wrap it in a call to form.runOnUiThread to ensure the right context for the blocks.